pkg/selector/generic/registry/registry.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package registry
+
+import (
+	"fmt"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/selector/generic"
+)
+
+type Registry map[string]SelectorFactory
+
+// SelectorFactory is a function that builds a selector.
+type SelectorFactory = func(selector v1alpha1.GenericSelectorSpec, option generic.Option) (generic.Selector, error)
+
+func Parse(registry Registry, spec v1alpha1.GenericSelectorSpec, option generic.Option) (generic.SelectorChain, error) {
+	selectors := make([]generic.Selector, 0, len(registry))
+	for name, factory := range registry {
```

Comment:
The order of iteration a map is not stable, and with the `replace` behaviours in SelectorChain for `ListOptions` and `ListFunc`, it might be the potential issue (if we build more selectors with overlapping `ListOptions` in the future.) 
Although obviously all the existed selectors do not have the overlapping `ListOptions`, it would work pretty well, how about appending some comments in here , or in `SelectorChain`, description the "replace logic" for the contributors.